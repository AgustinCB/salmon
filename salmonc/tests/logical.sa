true and true;
true and false;
false and true;
false and false;
true or true;
true or false;
false or true;
false or false;
