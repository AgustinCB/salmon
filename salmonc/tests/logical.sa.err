Constants: [Bool(true), Nil, Nil, Nil, Nil, Nil, Nil, Nil, Bool(false), Nil, Nil, Nil, Nil, Nil, Nil, Nil, String(0)]
Instructions: [Instruction { instruction_type: Constant(0), location: 0 }, Instruction { instruction_type: Constant(0), location: 0 }, Instruction { instruction_type: And, location: 0 }, Instruction { instruction_type: Constant(0), location: 1 }, Instruction { instruction_type: Constant(1), location: 1 }, Instruction { instruction_type: And, location: 1 }, Instruction { instruction_type: Constant(1), location: 2 }, Instruction { instruction_type: Constant(0), location: 2 }, Instruction { instruction_type: And, location: 2 }, Instruction { instruction_type: Constant(1), location: 3 }, Instruction { instruction_type: Constant(1), location: 3 }, Instruction { instruction_type: And, location: 3 }, Instruction { instruction_type: Constant(0), location: 4 }, Instruction { instruction_type: Constant(0), location: 4 }, Instruction { instruction_type: Or, location: 4 }, Instruction { instruction_type: Constant(0), location: 5 }, Instruction { instruction_type: Constant(1), location: 5 }, Instruction { instruction_type: Or, location: 5 }, Instruction { instruction_type: Constant(1), location: 6 }, Instruction { instruction_type: Constant(0), location: 6 }, Instruction { instruction_type: Or, location: 6 }, Instruction { instruction_type: Constant(1), location: 7 }, Instruction { instruction_type: Constant(1), location: 7 }, Instruction { instruction_type: Or, location: 7 }]
Locations: [Location { address: 0, line: 0 }, Location { address: 0, line: 1 }, Location { address: 0, line: 2 }, Location { address: 0, line: 3 }, Location { address: 0, line: 4 }, Location { address: 0, line: 5 }, Location { address: 0, line: 6 }, Location { address: 0, line: 7 }]
