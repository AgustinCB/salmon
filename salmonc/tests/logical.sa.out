0 - Bool(false)
1 - Bool(true)
2 - Bool(true)
3 - Bool(true)
4 - Bool(false)
5 - Bool(false)
6 - Bool(false)
7 - Bool(true)
